### DISCUSSION TITLE
How do i use Vercel Blob Storage in my Django Applicaton? If possible please also tell me a Way to use it via cURL.

### DISCUSSION THREAD

**Discussion by NotoriousArnav** [vercel/community#4682] [Help] [answered]
### Summary

I want to make a Blog Application for Learning some stuff. Now I need some storage, but i don't know how to use it. I am opting for Vercel Blob, because I am also using their platform to host the application and the database. I just want to know ow do i use Vercel Blob with cURL/Python to upload, modify, delete, retrieve files.

### Steps to Reproduce

Idk, but for https://0x0.st, in curl we use 
```sh
curl -X POST https://0x0.st -F"file=@file.ext"
```

How do i do the same for Vercel Blob?

**Comment by misaelnieto** [comment, marked as answer]
Hi @NotoriousArnav, I just released an experimental vercel blob wrapper for Python. 

https://github.com/misaelnieto/vercel-storage

Any comments will be greatly appreciated

**Reply by NotoriousArnav** [reply]
Yeah, I saw you code Just right now from my phone. Your implementation looks very good, but I have some questions.

Is Vercel Blob S3 compatible?
Can we make the uploaded file Publicly accessible?

**Reply by misaelnieto** [reply]
AFAIK, Vercel's blob is not  S3 compatible. It has it's own API. That's why I had to write the vercel-storage python package.

Also, according to the [API's documentation](https://vercel.com/docs/storage/vercel-blob/using-blob-sdk#put), all files you upload to Vercel Blob will be publicly available. Private uploads are planned, but not available currently.
